lab_3_generate_by_ngrams/main.py

```diff
@@ -41,6 +45,21 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or not text:
+            return None
+
+        tokens = []
+        for token in text.lower():
+            if token.isspace() and tokens and tokens[-1] != self._end_of_word_token:
```

Comment:
too dificult to understand the condition

Reply:
Changed